docs/source/tools/non-repudiation.rst

```diff
@@ -0,0 +1,78 @@
+once the feature is declared a stable part of Daml Connect Enterprise Edition. If you are interested in this feature, you are welcome to use it and
+give us feedback that will shape how this feature will ultimately come to be.
+
+Architecture
+~~~~~~~~~~~~
+
+The non-repudiation system consists of three components:
+
+- the non-repudiation middleware is a reverse proxy that sits in front of the Ledger API that verifies command signatures and forwards the signed command to the actual participant node
+- the non-repudiation API is a web server used by the operator to upload new certificates and verify repudiation claims
+- the non-repudiation client is a gRPC interceptor that can be used alongside any gRPC client on the JVM, including the official Java bindings, that will ensure that commands are signed with a given private key
+
+Running the server-side components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The server-side components are the middleware and the API. Both can be run as a single process by running the non-repudiation fat JAR provided as part of Daml Connect Enterprise Edition.
+
+Note that at the current stage you need to also have a PostgreSQL server running where signed commands will be persisted.
+
+The following example shows how to run the non-repudiation server components by connecting to a participant at localhost:6865 and proxying it to the 6866 port, using the given PostgreSQL instance to persist signed commands and certificates.
+
+.. code-block:: sh
+
+    java -jar /path/to/the/non-repudiation.jar --ledger-host localhost --ledger-port 6865 --proxy-port 6866 --jdbc url=jdbc:postgresql:nr,user=nr,password=nr
+
+For details on how to run them, please run the fat JAR with the ``--help`` command line option.
+
+Using the client
+~~~~~~~~~~~~~~~~
+
+The client is a gRPC interceptor which is available to Daml Connect Enterprise Edition users (hence, it's not available on Maven Central).
+
+The Maven coordinates for the library are `com.daml:non-repudiation-client`.
+
+The following example shows how to use this interceptor with the official Java bindings
+
+.. code-block:: java
+
+    PrivateKey key = readYourPrivateKey();
+    X509Certificate certificate = readYourX509Certificate();
+    NettyChannelBuilder builder = NettyChannelBuilder.forAddress(hostname, port);
+    builder.intercept(SigningInterceptor.signCommands(key, certificate));
+    DamlLedgerClient client = DamlLedgerClient.newBuilder(builder).build();
+    client.connect();
+
+Non-repudiation over the HTTP JSON API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note that the current status is that non-repudiation applies exclusively to Ledger API messages: if you want to use the HTTP JSON API you will need to
+run your own HTTP JSON API server and start it with a certificate that will be used to sign every command issued by the HTTP JSON API to the participant.
+This means that running the HTTP JSON API with non-repudiation as a ledger operator is currently not supported.
```

Comment:
I’d rephrase this slightly. It is supported and you can come up with cases where this is sensible, e.g., if you don’t expose it publically. The crucial point is that you don’t get non-repudiation for the leg user <-> http json API.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/9288/commits/686fe9c37e8ec0939e35bf477aa1c7c1a7d3af8c